editors/emacs/Portfile

```diff
@@ -142,9 +145,20 @@ if {$subport eq "emacs-devel" || $subport eq "emacs-app-devel"} {
 
     livecheck.type none
 
+    # Emacs 30 removes libjansson in favour of native JSON support.
+    configure.args-delete   --with-json
+    depends_lib-delete  port:jansson
+    configure.args-append   --with-native-compilation=no
+
+    # TODO: Put this somewhere better. See https://github.com/emacs-mirror/emacs/commit/4c6f45fa8eef1a15d5790c1f3d3e608b548015db#diff-731c852b7c27c6ee90c44c871e2988ecfc2d225bc720156d7a47c1d563c2683aR59
+    configure.args-append   --disable-gc-mark-trace
+
     if {[variant_isset nativecomp]} {
         notes "emacs devel subports don't always keep compatibility for native\
  compiled files. Better to cleanup your ~/.emacs.d/.local/cache/eln"
+
+        configure.args-delete   --with-native-compilation=no
+        configure.args-append   --with-native-compilation=aot
```

Comment:
this could/should use `configure.args-replace` instead

Reply:
Will get to it, going to wait for @drkp's feedback for the other item in this file (do both at once).

Reply:
Just finish it up. None of the maintainers had responded in over three weeks, so don't wait. Just get these last things finalized and I'll merge it.

Reply:
Alrighty, will get on it in an hour or two.